services/app-web/src/pages/StateSubmission/StateSubmissionForm.tsx

```diff
@@ -192,13 +191,19 @@ export const StateSubmissionForm = (): React.ReactElement => {
             if (!updatedSubmission) {
                 setShowPageErrorMessage(true)
                 console.log('Failed to update form data', updateResult)
+                recordJSException(
+                    `StateSubmissionForm: Apollo error reported. Error message: Failed to update form data ${updateResult}`
+                )
                 return new Error('Failed to update form data')
             }
 
             return updatedSubmission
         } catch (serverError) {
             setShowPageErrorMessage(true)
-            return serverError
+            recordJSException(
+                `StateSubmissionForm: Apollo error reported. Error message: ${serverError.message}`
+            )
+            throw new Error(serverError)
```

Comment:
Throwing this error because `updateDraftHealthPlanPackage` is called in child components that has a `catch` block.

Reply:
I’m a little confused. We are returning a new error in one place but throwing in another in this same function? I think we should treat this function like one of our dependency wrappers, we are calling the apollo API but we should be able to return an | Error like we do everywhere else. 

That can wait until we extract all our apollo calls into a wrapper module, I know we have a ticket for doing that for the fetch API so we can do the proto unzipping behind the same facade.